jsonnet/prometheus.jsonnet

```diff
@@ -209,6 +209,7 @@ local namespacesRole =
     serviceMonitorApiserver:: {},
     serviceMonitorKubeScheduler:: {},
     serviceMonitorKubeControllerManager:: {},
+    serviceMonitorCoreDNS:: {},
```

Comment:
this is just a small consolidation which removes complexity from `build-jsonnet.sh`. Currently, we `rm -f ...` unneeded assets which come from kube-prometheus.

Instead of duplicating `rm -f ...` statements in `build-jsonnet.sh` for the userworkload use case, I shadow those assets in jsonnet using `::`. That way they won't be rendered in the first place and we can remove the `rm -f ...` statements for good, see https://github.com/openshift/cluster-monitoring-operator/pull/499/files#diff-c92e0d4cd5002a9ca161eb6d594f5987L52.